app/src/main/java/com/example/travellens/Filter.java

```diff
@@ -27,6 +47,48 @@ public static Filter getInstance() {
         return instance;
     }
 
+    // complex filtering algorithm
+    public List<Post> getPostsByFilteringMostRecommended(List<String> types, List<Post> currentPosts) {
+        PriorityQueue<WeightedPost> weightedPostPriorityQueue = new PriorityQueue<WeightedPost>(new Comparator<WeightedPost>() {
+            @Override
+            public int compare(WeightedPost thisPost, WeightedPost otherPost) {
+                // implement custom comparator for my weighted post
+                return Double.compare(otherPost.weight, thisPost.weight);
+            }
+        });
+
+        // iterate through each post and see how many filters it contains, calculate those weights
+        for (int post = 0; post < currentPosts.size(); post++) {
+            Post currentPost = currentPosts.get(post);
+            int weightsForPost = 0;
+            // Weights calculated by currentPosts.size() * (types.size() - type))
+            for (int type = 0; type < types.size(); type++) {
```

Comment:
Same here.. update the for loop running index to typeIndex and update the comment to calculate weight to types.size() - typeIndex